### PR TITLE
Add HTML proofer to Jekyll tests

### PR DESCRIPTION
**Pull request by jdavis**
As mentioned in the commit message, this was inspired by https://github.com/neovim/neovim.github.io/pull/84.

Pull it locally and give it a go. Before this is merged, we should fix up the links, so far these are the ones it finds that are bad:

```
Checking 798 external links...
Ran on 9 files!


- ./_site/index.html
  *  linking to internal hash #backup that does not exist
  *  External link https://coinkite.com failed: 503 No error
  *  External link https://coinkite.com/settings/pws failed: 503 No error
  *  External link https://docs.compose.io/security/mongohq-two-factor-authentication.html failed: 404 No error
  *  External link https://www.maxcdn.com/one/tutorial/tutorials/enabling-2-step-verification/ failed: 404 No error
  *  External link https://www.mail.yahoo.com/ failed: got a time out
  *  External link https://ssltrust.com.au/ failed: 0 Peer certificate cannot be authenticated with given CA certificates
  *  External link https://www.cacert.org/ failed: 0 Peer certificate cannot be authenticated with given CA certificates
  *  External link https://justcoin.com/ failed: 302 Number of redirects hit maximum amount
  *  External link https://blog.nearlyfreespeech.net/2014/02/28/price-cuts-more-se failed: 404 No error
  *  External link http://help.ea.com/en/article/origin-login-verification-information/ failed: 302 Number of redirects hit maximum amount
  *  External link http://mojang.com failed: 503 No error
  *  External link https://www.netflix.com/home failed: 301 Number of redirects hit maximum amount
  *  External link http://na.square-enix.com/en_all/sso/check?mret=http%3A%2F%2Fna.square-enix.com%2F failed: 302 Number of redirects hit maximum amount
  *  External link http://www.looker.com/docs/admin/two-factor-authentication failed: 404 No error
- ./_site/providers/index.html
  *  External link https://www.SecureAccessTechnologies.com/ failed: 400 No error
rake aborted!
HTML-Proofer found 16 failures!
```

### CODE REVIEW

File: _data/domains.yml [hunk 1]
```diff
@@ -55,11 +55,11 @@ websites:
       doc: http://blog.dnsimple.com/2012/08/protect-your-dnsimple-account-with-two-factor-authentication-from-authy/
 
     - name: Domain.com
-      url: https://domain.com
+      url: http://www.domain.com/
```
Comment (mxxcon) [inline _data/domains.yml:58]: https://domain.com/ still works for me..

Comment (jdavis) [inline _data/domains.yml:58]: The proofer failed on it apparently. I can change it back and see if it still fails.

**Comment by mxxcon** [comment]
https://coinkite.com works for me. Looks like it was CloudFlare hickup.
https://ssltrust.com.au fails cert because it's for https://www.ssltrust.com.au
https://www.cacert.org fails because it's not part of browser-standard root CA store.
https://justcoin.com works for me.
http://help.ea.com/en/article/origin-login-verification-information/ works fine too.
Looker docs url should be http://www.looker.com/docs/admin/security/two-factor-authentication

**Comment by jdavis** [comment]
@mxxcon I've fixed all of the errors and am ignoring the 302 errors: https://travis-ci.org/jdavis/twofactorauth/builds/46445871

**Comment by jdavis** [comment]
Nothin? Okay then.

**Pull request merged**